### PR TITLE
Reit demo/devidend

### PR DESCRIPTION
**Pull request by Electr1Xx**

### CODE REVIEW

File: envs/local.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   NODE_ENV: '"default"',
-  HORIZON_SERVER: '"http://localhost:8000/_/api/"',
-  FILE_STORAGE: '"http://localhost:8000/_/storage/api"',
-  NETWORK_PASSPHRASE: '"TokenD Developer Network"',
+  HORIZON_SERVER: '"https://api.reit.tokend.io"',
```
Comment (letehaha) [inline envs/local.js]: This config is incorrect. Check for right here: https://reit.tokend.io/static/env.js

File: src/i18n/en.json [hunk 2]
```diff
@@ -1015,6 +1016,28 @@
     "terms-page": {
       "go-back-btn": "Go back",
       "title": "Terms & conditions"
+    },
+
+    "dividend-form": {
+      "title": "Dividends",
+      "asset": "Asset",
+      "loading-msg": "Loading",
+      "loading-error-msg":"There was an error while loading.",
```
Comment (letehaha) [inline src/i18n/en.json]: Point at the end is disallowed

File: src/vue/modules/dividend-form/scss/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+$movements-table-cell-width: 10rem;
```
Comment (letehaha) [inline src/vue/modules/dividend-form/scss/_variables.scss]: Useless file

File: src/vue/modules/dividend-form/wrappers/asset.record.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import { ASSET_POLICIES } from '@tokend/js-sdk'
```
Comment (letehaha) [inline src/vue/modules/dividend-form/wrappers/asset.record.js]: In the record, you need to determine only values that you used in the component

File: src/vue/modules/dividend-form/index.vue [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import EmailGetter from '@/vue/common/EmailGetter'
+import { types } from './store/types'
+
+import { FEE_TYPES, PAYMENT_FEE_SUBTYPES, Wallet } from '@tokend/js-sdk'
+import { api, initApi } from './_api'
+import { AssetRecord } from './wrappers/asset.record'
+import required from 'vuelidate/src/validators/required'
+import { amount, maxDecimalDigitsCount, noMoreThanAvailableOnBalance } from '../../../validators'
+import * as base from '@tokend/js-sdk/lib/base/operations/payment_builder'
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Did you check for examples? `import { base } from '@tokend/js-sdk'` is enough

File: src/vue/modules/dividend-form/index.vue [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+import { mapActions, mapGetters, mapMutations } from 'vuex'
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import EmailGetter from '@/vue/common/EmailGetter'
+import { types } from './store/types'
+
+import { FEE_TYPES, PAYMENT_FEE_SUBTYPES, Wallet } from '@tokend/js-sdk'
+import { api, initApi } from './_api'
+import { AssetRecord } from './wrappers/asset.record'
+import required from 'vuelidate/src/validators/required'
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Look at examples. This import is ugly and incorrect

File: src/vue/modules/dividend-form/index.vue [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+import { amount, maxDecimalDigitsCount, noMoreThanAvailableOnBalance } from '../../../validators'
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Here are more than 80 symbols in the line. Do you have an ESLint issue about it? If not, config your IDE correctly

File: src/vue/modules/dividend-form/index.vue [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+import * as base from '@tokend/js-sdk/lib/base/operations/payment_builder'
+import { Balance } from './wrappers/balance'
+import { MathUtil } from '../../../js/utils/math.util'
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Use alias

File: src/vue/modules/dividend-form/index.vue [hunk 9]
```diff
@@ -0,0 +1,530 @@
...
+        this.$emit(EVENTS.operationSubmitted)
+      } catch (e) {
+        ErrorHandler.process(e)
+      }
+      this.enableForm()
+    },
+    async getSigners () {
+      this.isSignersLoaded = false
+      try {
+        const { data } = await api().getWithSignature(`/${HORIZON_VERSION_PREFIX}/balances`, {
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Move this loading into store

File: src/vue/modules/dividend-form/index.vue [hunk 10]
```diff
@@ -0,0 +1,530 @@
...
+        },
+      },
+    }
+  },
+  async created () {
+    initApi(this.wallet, this.config)
+
+    this.setAccountId(this.wallet.accountId)
+    await this.loadBalances()
+    this.isInitialized = true
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Move it to the bottom of the `created()` hook

File: src/vue/modules/dividend-form/index.vue [hunk 11]
```diff
@@ -0,0 +1,530 @@
...
+              type="submit"
+              class="app__button-raised"
+              :disabled="formMixin.isDisabled || !balanceHolders.length"
+              form="dividend-form"
+            >
+              {{ 'dividend-form.dividend-btn' | globalize }}
+            </button>
+            <form-confirmation
+              v-if="formMixin.isConfirmationShown"
+              @ok="hideConfirmation() || submit()"
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Move `hideConfirmation()` to the end of `submit` method. Also, you need to define a new variable that will tell to component about form submitting and pass it here via `isPending`. Look how this prop works for formConfirmation component

File: src/vue/modules/dividend-form/index.vue [hunk 12]
```diff
@@ -0,0 +1,530 @@
...
+    countIssuedValue () {
+      if (+this.ownedUserBalance !== 0) {
+        return MathUtil
+          .subtract(+this.form.ownedAsset.issued, +this.ownedUserBalance)
+      } else {
+        return +this.form.ownedAsset.issued
+      }
+    },
+    async getAccountId (id) {
+      const { _rawResponse } = await api().get(`/balances/${id}/account`, {})
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Use store to make requests

File: src/vue/pages/Movements.vue [hunk 13]
```diff
@@ -108,10 +126,12 @@ import { ErrorHandler } from '@/js/helpers/error-handler'
 import { AssetRecord } from '@/js/records/entities/asset.record'
 
 import config from '../../config'
+import DividendFormModule from '../modules/dividend-form/index'
```
Comment (letehaha) [inline src/vue/pages/Movements.vue]: Use alias

File: src/vue/modules/dividend-form/index.vue [hunk 14]
```diff
@@ -0,0 +1,530 @@
...
+        this.balanceHolders
+          .map(async item => {
+            item.accountId = await this.getAccountId(item.id)
+            return item
+          })
+      )
+    },
+    deleteOwnedBalance (balance) {
+      if (balance.accountId === this.accountId) {
+        this.ownedUserBalance = this.getFullBalanceHolder(balance)
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: What did you want to return here for your filter [here](https://github.com/tokend/web-client/pull/137/files#diff-23af4977e26eed3d7cb9ce46a739a981R306)?

File: src/vue/modules/dividend-form/index.vue [hunk 15]
```diff
@@ -0,0 +1,530 @@
...
+        const updatedAsset = this.assets
+          .find(item => item.code === this.form.asset.code)
+        const updatedOwnedAsset = this.ownedAssets
+          .find(item => item.code === this.form.ownedAsset.code)
+        this.form.asset = updatedAsset || this.assets[0]
+        this.form.ownedAsset = updatedOwnedAsset || this.ownedAssets[0]
+      }
+    },
+    async loadAssets () {
+      const { data } = await api().get(`/${HORIZON_VERSION_PREFIX}/assets`, {})
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Use store

File: src/vue/modules/dividend-form/index.vue [hunk 16]
```diff
@@ -0,0 +1,530 @@
...
+      return this.balanceHolders
+        .filter(holder =>
+          this.getFullBalanceHolder(holder) >= this.config.minAmount
+        )
+    },
+    getFullBalanceHolder (holder) {
+      return MathUtil
+        .add(+holder.available, +holder.locked)
+    },
+    async buildPaymentOperation (holder, fees) {
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Useless async

File: src/vue/modules/dividend-form/index.vue [hunk 17]
```diff
@@ -0,0 +1,530 @@
...
+      this.ownedAssets = assets
+        .filter(item => item.owner === this.accountId)
+    },
+    async getTransferOperations () {
+      let operations = []
+      let balanceHolders = this.filterPositiveBalance()
+      await Promise.all(
+        balanceHolders.map(async holder => {
+          const fees = await this.getFees(holder)
+          const data = await this.buildPaymentOperation(holder, fees)
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Useless await

File: src/vue/modules/dividend-form/index.vue [hunk 18]
```diff
@@ -0,0 +1,530 @@
...
+        }),
+      ])
+      return {
+        source: senderFees,
+        destination: recipientFees,
+      }
+    },
+    async loadPaymentFee ({ asset, amount, account, subtype }) {
+      try {
+        const { data } = await api().getWithSignature(`/${HORIZON_VERSION_PREFIX}/accounts/${account}/calculated_fees`, {
```
Comment (letehaha) [inline src/vue/modules/dividend-form/index.vue]: Use store

File: src/vue/modules/dividend-form/index.vue (hunk 14 above)
Comment (Electr1Xx) [reply src/vue/modules/dividend-form/index.vue]: If owner balance exist, I calculate balance and write to a variable and return false else return true

**Pull request merged**